src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -729,12 +759,31 @@ int implGCMCrypt(ByteBuffer src, ByteBuffer dst) {
             int len;
 
             if (src.hasArray() && dst.hasArray()) {
-                ByteBuffer ct = (encryption ? dst : src);
-                len = GaloisCounterMode.implGCMCrypt(src.array(),
-                    src.arrayOffset() + src.position(), srcLen,
-                    ct.array(), ct.arrayOffset() + ct.position(),
-                    dst.array(), dst.arrayOffset() + dst.position(),
-                    gctr, ghash);
+                byte[] array;
+                if (encryption) {
+                    array = dst.array();
```

Comment:
You could factor out lines 764 and 770 by changing line 762 to
`byte[] array = encryption ? dst.array() : src.array();`

Reply:
That was intentional since 763 checks the encryption boolean, I can define 'array' in that condition instead of having two conditions for the same thing